pdu.c

```diff
@@ -920,134 +779,108 @@ EXPORT_DEF const char * pdu_parse(char ** pdu, size_t tpdu_length, char * oa, si
 			break;
 		}
 		if (reserved) {
-			*pdu -= 2;
-			return "Reserved DCS value";
+			return -1;
+		}
+		if (alphabet == -1) {
+			return -1;
 		}
 	}
-	if (alphabet == -1) {
-		*pdu -= 2;
-		return "Unsupported DCS value";
-	}
-
-	ts = pdu_parse_timestamp(pdu, &pdu_length);
-	*msg_enc = pdu_dcs_alphabet2encoding(alphabet);
-	if (ts < 0) {
-		return "Can't parse Timestamp";
-	}
-
-	udl = pdu_parse_byte(pdu, &pdu_length);
-	if (udl < 0) {
-		return "Can't parse UDL";
+	if (alphabet == PDU_DCS_ALPHABET_8BIT) {
+		// TODO: What to do with binary messages? Are there any?
+		// Return an error as it is dangerous to forward the raw binary data as text
```

Comment:
Looks like you're dropping all hints in this function why we failed parsing (all returning -1 now).

Might be worse for debugging problems later on..